### PR TITLE
fix(styles): fix overflow styling on Radio Group and Checkbox and align styles with Pattern Library

### LINKED ISSUE #627: Checkbox wrapping issue
checkbox with label above the visual checkbox

Checkboxes have an issue when the label needs to wrap, the label content appears before the checkbox.

### PR DESCRIPTION
**Pull request by yhafez**
Closes #627 
Relates to #949 
Relates to #1034 

Relates to Walnut [#5739](https://github.com/dequelabs/walnut/issues/5739)

Current behavior:
https://github.com/dequelabs/cauldron/assets/60084578/b94f6133-0c7c-4368-9277-6736618301f9

---

With .Checkbox, .Radio `flexwrap` changed to `nowrap`:
https://github.com/dequelabs/cauldron/assets/60084578/ee399c8d-88c7-4b37-be61-2e2b4403bb21
no align-item self

w/ 4th option under wrapping line:
https://github.com/dequelabs/cauldron/assets/60084578/4dbaa581-63a3-41e5-9a02-d0033e09c6e0

---

With .Checkbox, .Radio `flexwrap` changed to `nowrap` and .Radio__overlay given `align-self` of `flex-start` (implemented):
https://github.com/dequelabs/cauldron/assets/60084578/6142d258-e011-451f-9427-bff88b6bcee9
align-item self

w/ 4th option under wrapping line:
https://github.com/dequelabs/cauldron/assets/60084578/fbe68121-6973-4831-af4b-12316976020d

### CODE REVIEW

**Comment by yhafez** [comment]
> Checkbox/radio share styles, we cannot make a change to one without changing both, see below screenshot:
> 
> cauldron checkbox with description to the left of the checkbox label and checkbox

Addressed!

**Comment by scurker** [comment]
After a quick zoom meeting with @yhafez and @dequelabs/cauldron-design, we settled on the following:

- Make sure the final designs for RadioGroup and Checkbox match the [Pattern Library](https://preview.uxpin.com/520e80315786a67ba43a3ca50598e6d69782627c#/pages/161871161/simulate/no-panels?mode=i)
  - This should partially address #949 and fully address #1034 
- Change the outer wrapper to use a class with something like `Checkbox__wrap` and `RadioGroup__wrap`

File: packages/styles/forms.css [hunk 1]
```diff
@@ -16,6 +16,7 @@
   --field-label-text-color: var(--gray-90);
   --field-label-description-text-color: var(--gray-60);
   --input-min-width: 250px;
+  --focus-glow: rgba(181, 26, 209, 1);
```
Comment (scurker) [inline packages/styles/forms.css:19]: We want to pull from global vars here:
https://github.com/dequelabs/cauldron/blob/720bc9bd5db3eab975d0c6d80639c8614f3899a4/packages/styles/variables.css#L41-L42

We also already have `focus-glow` defined in vars so we probably need to rescope the var here. Ideally following our guidelines it should be descriptive something like:

```css
--field-focus-ring-color: var(--focus-light);
```

With the alternative set for dark mode as well.

File: packages/styles/forms.css [hunk 2]
```diff
@@ -159,10 +161,12 @@ textarea.Field--has-error:focus:hover,
 
 .Error {
   color: var(--error);
+  border-top: 1px solid var(--error);
   text-align: left;
   font-size: var(--text-size-smallest);
-  font-weight: var(--font-weight-normal);
-  padding-left: var(--space-half);
+  margin-top: var(--space-half);
+  margin-left: calc(24px + 2px + var(--space-half));
```
Comment (scurker) [inline packages/styles/forms.css:168]: I know this was copy pasta, but we should update this to use `--icon-size`:

```suggestion
  margin-left: calc(var(--icon-size) + 2px + var(--space-half));
```

File: packages/styles/forms.css [hunk 3]
```diff
@@ -395,5 +397,11 @@ textarea.Field--has-error:focus:hover,
 .cauldron--theme-dark .Radio__overlay--disabled,
 .cauldron--theme-dark .Checkbox__overlay--disabled.Icon--checkbox-checked,
 .cauldron--theme-dark .Checkbox__overlay--disabled.Icon--checkbox-unchecked {
-  color: #4c99a9;
+  color: #5d676f;
```
Comment (scurker) [inline packages/styles/forms.css:400]:
```suggestion
  color: var(--stroke-dark);
```

File: packages/styles/forms.css [hunk 4]
```diff
@@ -395,5 +397,11 @@ textarea.Field--has-error:focus:hover,
...
+}
+
+.Checkbox__overlay.Checkbox__overlay--focused,
+.Radio__overlay.Radio__overlay--focused {
+  border-radius: 5px;
+  box-shadow: 0 0 0 2px var(--focus-glow);
```
Comment (scurker) [inline packages/styles/forms.css:406]: It looks like this was defined twice.

File: packages/styles/forms.css [hunk 5]
```diff
@@ -31,6 +32,7 @@
   --field-required-text-color: var(--white);
   --field-label-text-color: var(--white);
   --field-label-description-text-color: rgba(255, 255, 255, 0.8);
+  --focus-glow: rgba(235, 148, 255, 1);
```
Comment (scurker) [inline packages/styles/forms.css:35]: Same thing as above.

File: packages/styles/forms.css [hunk 6]
```diff
@@ -197,13 +206,9 @@ textarea.Field--has-error:focus:hover,
...
-.Checkbox {
-  margin-bottom: var(--space-smallest);
-}
-
-.Checkbox + .Checkbox,
-.Radio + .Radio {
-  margin-top: var(--space-smallest);
+.Checkbox__wrap:last-of-type,
+.Radio__wrap:last-of-type {
+  margin-bottom: 0;
```
Comment (scurker) [inline packages/styles/forms.css:211]: Clever!

File: packages/styles/forms.css [hunk 7]
```diff
@@ -364,7 +363,10 @@ textarea.Field--has-error:focus:hover,
   color: #3c7aae;
 }
 
-.Checkbox__overlay--disabled.Icon--checkbox-checked,
+.Checkbox__overlay--disabled.Icon--checkbox-checked {
+  color: #78a6d8;
```
Comment (scurker) [inline packages/styles/forms.css:367]: Following our [variable guidelines](https://cauldron.dequelabs.com/style-guidelines#variables) this should be either inheriting from a var or specifying a fallback value. I couldn't find this defined anywhere else so we should probably specify that as fallback value:

```css
color: var(--field-checkbox-disabled-color, #78a6d8)
```

File: packages/styles/forms.css [hunk 8]
```diff
@@ -364,7 +363,10 @@ textarea.Field--has-error:focus:hover,
...
+}
+
 .Checkbox__overlay--disabled.Icon--checkbox-unchecked {
   color: #ccc;
```
Comment (scurker) [inline packages/styles/forms.css:371]: While we're here, we should update this to use our global variable (similar to above):

```css
color: var(--field-checkbox-unchecked-disabled-color, var(--gray-40))
```
```

**Review by scurker — CHANGES_REQUESTED** [review]
Good work! Just a few nits to fix.

**Comment by scurker** [comment]
Update from @dequelabs/cauldron-design 

> There should be a global var `accent-primary-disabled` set to `#78a6d8` as we will be using it elsewhere.

**Comment by bobbyomari** [comment]
@yhafez here are my observations...

### checkbox dark theme:
- the active fill on touch down should be `accent-light` @ 25% opacity
- the active fill on touch down in error state should be `accent-danger` @ 25% opacity
- the `.Field_labelDescription` should be color `accent-light`
- Set `margin-top: var(--space-smallest);` for `.error` on checkbox with description. I think it strikes a better visual balance with how the code is structured (see attached screenshots, the one with more spacing is 8px). Looks fine with what it's at on
- If label is being used with checkboxes, can we set the field margin-bottom to `var(--space-smallest)`? This also comes to play with radios.

### checkbox light theme:
- the active fill on touch down should be `accent-primary` @ 25% opacity
- the active fill on touch down in error state should be `error` @ 25% opacity
- same as dark theme... set `margin-top: var(--space-smallest);` for `.error` on checkbox with description

### radio dark theme:
- it appears the radio focus is not circular, it's a rectangle
- `.Field_label` `padding-bottom` should be `var(--space-smallest)` 
- related to above: should we use `margin-bottom` instead of `padding-bottom`?
- like dark checkboxes, the active fill on touch down should be the same
- on inline layout, remove `margin-bottom` from `.Radio_wrap`

### radio light theme:
- same radio focus issue as dark theme
- same `.Field_label` request as dark theme
- like light checkboxes, the active fill on touch down should be the same
- same inline layout `margin-bottom` change as dark theme radios

### current implementation with `margin-top: 4px;`
image

### updated implementation with `margin-top: 8px;`
image

File: packages/styles/forms.css (hunk 4 above)
Comment (yhafez) [reply packages/styles/forms.css:406]: Ah, this was originally supposed to include `.cauldron--theme-dark`, but forgot to add that, though it looks like the way the CSS variables work, it's not necessary after all. Removed!

**Comment by bobbyomari** [comment]
I already sent these changes to @yhafez but wanted to document them here as well for a 2nd PR review...

### Checkboxes
- I noticed on the error state, when the checkbox is empty, the active fill is correct, but when the error checkbox is checked, the active fill stays red. Is there way to make this change to red active fill **only when the checkbox is unchecked and in an error state?** This applies to light and dark theme. 

### Radios
- The label for "do you like pizzas?" is now no longer `font-weight: 500;`
- Some discrepancies between Cauldron in UXPin vs development...
-- The space between the radio label and description should be `var(--space-half);`
-- The space between radio groups should be `var(--space-small);`
-- This helps create some better visual balance between what description belongs to what radio field. Below are screenshots to share what that change looks like.

### Radio with long description
image

### Radio with short description
image

File: docs/pages/components/Checkbox.mdx [hunk 9]
```diff
@@ -86,6 +86,48 @@ import { Checkbox } from '@deque/cauldron-react'
...
+    value="1"
+  />
+  <Checkbox
+    disabled
+    checked
+    id="checkbox-list-checkbox-disabled-checked"
+    label="Unchecked Disabled Checkbox"
+    name="disabled-checkbox-list-checked"
+    value="1"
+  />
```
Comment (scurker) [inline docs/pages/components/Checkbox.mdx:120]: It looks like these 2 are swapped

File: packages/styles/forms.css [hunk 10]
```diff
@@ -14,7 +14,21 @@
...
+    var(--accent-primary) 20%,
+    transparent
+  );
+  --icon-error-active-color: color-mix(in srgb, var(--error) 20%, transparent);
+  --icon-checked-color: var(--accent-primary);
+  --icon-checked-disabled-color: var(--accent-primary-dsiabled, #78a6d8);
+  --icon-unchecked-disabled-color: var(--gray-40);
+  --icon-focus-color: var(--focus-light);
+  --error-text-color: var(--error);
+  --error-border-color: var(--error);
```
Comment (scurker) [inline packages/styles/forms.css:31]: Can we prefix these with `field`?

File: packages/styles/forms.css [hunk 11]
```diff
@@ -30,7 +44,20 @@
...
+    in srgb,
+    var(--accent-danger) 20%,
+    transparent
+  );
+  --icon-checked-color: var(--accent-light);
+  --icon-checked-disabled-color: var(--stroke-dark);
+  --icon-unchecked-disabled-color: var(--stroke-dark);
+  --icon-focus-color: var(--focus-dark);
+  --error-text-color: var(--error);
+  --error-border-color: var(--error);
```
Comment (scurker) [inline packages/styles/forms.css:60]: See comment above.

File: packages/styles/forms.css [hunk 12]
```diff
@@ -14,7 +14,21 @@
...
   --field-required-text-color: var(--gray-60);
   --field-label-text-color: var(--gray-90);
+  --field-label-error-text-color: var(--error);
   --field-label-description-text-color: var(--gray-60);
+  --icon-inactive-color: var(--gray-90);
+  --icon-active-color: color-mix(
+    in srgb,
+    var(--accent-primary) 20%,
+    transparent
+  );
```
Comment (scurker) [inline packages/styles/forms.css:24]: GIven that this was only recently introduced we should define a static value so we don't accidentally introduce color contrast issues to users who do not yet support this feature.

**Review by scurker — CHANGES_REQUESTED** [review]
Looking good! Just a last few code changes needed.

File: docs/pages/components/Checkbox.mdx (hunk 9 above)
Comment (yhafez) [reply docs/pages/components/Checkbox.mdx:120]: Fixed

File: packages/styles/forms.css (hunk 12 above)
Comment (yhafez) [reply packages/styles/forms.css]: Just went with the rgba values for now as per [our discussion on slack](https://deque.slack.com/archives/C5MUFQ7UM/p1689889583970209)

File: packages/styles/forms.css [hunk 13]
```diff
@@ -197,29 +236,41 @@ textarea.Field--has-error:focus:hover,
...
   color: var(--field-label-text-color);
+  margin-bottom: var(--space-half);
   font-size: var(--text-size-small);
   font-weight: var(--font-weight-medium);
-  padding-bottom: var(--space-half);
+  cursor: default;
+}
+
+.Field__label + .Checkbox__wrap,
+.Field__label + div[role='radiogroup'] {
```
Comment (scurker) [inline packages/styles/forms.css:262]: What if `role="radiogroup"` isn't a div?

File: packages/styles/forms.css [hunk 14]
```diff
@@ -118,6 +137,10 @@ textarea.Field--has-error,
   color: var(--field-border-color-error);
 }
 
+.Icon--checkbox-unchecked.Checkbox__overlay:active.Field--has-error:not(.Checkbox__overlay--disabled) {
```
Comment (scurker) [inline packages/styles/forms.css:140]:
```suggestion
.Checkbox__overlay:active.Field--has-error:not(.Checkbox__overlay--disabled) {
```

This feels like an overly specific selector, can we not just use `.Checkbox__overlay`?

**Review by scurker — COMMENTED** [review]
Code LGTM, 2 questions. Let's get @dequelabs/cauldron-design to give a thumbs up too.

File: packages/styles/forms.css (hunk 13 above)
Comment (yhafez) [reply packages/styles/forms.css]: Used `.Field__label + [role^='radiogroup']` instead~

File: packages/styles/forms.css (hunk 14 above)
Comment (yhafez) [reply packages/styles/forms.css:140]: Ah in this case, @bobbyomari mentioned to me that it should only show the red active fill state when checking the unchecked box, not when unchecking the checked box

Comment (scurker) [reply packages/styles/forms.css:140]: If that is the case, we should be checking the checked state and not the class.

Comment (scurker) [reply packages/styles/forms.css:140]: Something like `.Checkbox__overlay:not(:checked):active...`

**Review by scurker — APPROVED** [review]
Good work @yhafez!

File: packages/styles/forms.css [hunk 15]
```diff
@@ -158,11 +181,14 @@ textarea.Field--has-error:focus:hover,
...
+  color: var(--field-error-text-color);
+  border-top: 1px solid var(--field-error-border-color);
   text-align: left;
-  font-size: var(--text-size-smallest);
   font-weight: var(--font-weight-normal);
-  padding-left: var(--space-half);
+  font-size: var(--text-size-smallest);
+  margin-top: var(--space-half);
+  margin-left: calc(var(--icon-size) + 2px + var(--space-half));
+  padding: var(--space-half) 0;
```
Comment (tbusillo) [inline packages/styles/forms.css:191]: @yhafez This is the change that is doing this.

**Pull request merged**